tests/optim/test_oss_adascale.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pylint: disable=missing-module-docstring
+# pylint: disable=missing-class-docstring
+# pylint: disable=missing-function-docstring
+
+""" Test AdaScale with OSS. """
+
+from statistics import mean
+import tempfile
+
+import numpy as np
+import pytest
+import torch
+from torch import Tensor
+import torch.distributed as dist
+import torch.multiprocessing as mp
+from torch.nn import Linear, Sequential
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.optim import SGD
+
+from fairscale.optim import OSS, AdaScale
+from fairscale.utils.golden_testing_data import adascale_test_data
+from fairscale.utils.testing import skip_if_single_gpu
+
+
+def _dist_init(rank, world_size, tempfile_name, backend):
+    url = "file://" + tempfile_name
+    dist.init_process_group(init_method=url, backend=backend, rank=rank, world_size=world_size)
+    torch.cuda.set_device(rank)
+
+
+def _test_basic_func(rank, world_size, tempfile_name, test_case, oss, model=None):
+    _dist_init(rank, world_size, tempfile_name, backend="nccl")
+
+    if model is None:
+        model = Linear(2, 2, bias=False)
+    model.to("cuda")
+    model = DDP(model, device_ids=[rank])
+    if oss:
+        # For now, we can only wrap AdaScale over OSS. If we do it the other way around,
+        # AdaScale needs to take different parameter types, i.e. the parameter list, etc.
+        optim = AdaScale(OSS(model.parameters(), SGD, lr=0.1))
+    else:
+        optim = AdaScale(SGD(model.parameters(), lr=0.1))
+
+    if "input" in test_case:
+        inputs = [test_case["input"]]
+    else:
+        inputs = test_case["inputs"]
```

Comment:
when does this happen?

Reply:
This is just two different ways to specifying the input(s). In one case, it is a single input, in the other case, it is a list of inputs for multiple iterations. Seems to me it make the input data more readable this way (in the golden data file.)